lockc/src/runc.rs

```diff
@@ -0,0 +1,589 @@
+                // After handling the option, start parsing the next argument.
+                continue;
+            }
+
+            match opt_parsing_action {
+                OptParsingAction::NoPositional => {}
+                OptParsingAction::Skip => {
+                    opt_parsing_action = OptParsingAction::NoPositional;
+                    continue;
+                }
+                OptParsingAction::Bundle => {
+                    container_bundle_o = Some(arg);
+                    opt_parsing_action = OptParsingAction::NoPositional;
+                    continue;
+                }
+            }
+            match arg_parsing_action {
+                ArgParsingAction::None => {}
+                ArgParsingAction::ContainerId => {
+                    container_id_o = Some(arg);
+                    arg_parsing_action = ArgParsingAction::None;
+                    continue;
+                }
+            }
+
+            match arg.as_str() {
+                "checkpoint" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "create" => {
+                    arg_parsing_action = ArgParsingAction::ContainerId;
+                    container_action = ContainerAction::Create;
+                }
+                "delete" => {
+                    arg_parsing_action = ArgParsingAction::ContainerId;
+                    container_action = ContainerAction::Delete;
+                }
+                "events" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "exec" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "kill" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "pause" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "ps" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "restore" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "resume" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "run" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "start" => {
+                    arg_parsing_action = ArgParsingAction::ContainerId;
+                }
+                "state" => arg_parsing_action = ArgParsingAction::ContainerId,
+                "update" => arg_parsing_action = ArgParsingAction::ContainerId,
+                _ => {}
+            }
+        }
+
+        match container_action {
+            ContainerAction::Other => {
+                debug!("other container action");
+                if let Some(v) = container_id_o {
+                    let container_key = crate::hash(&v)?;
+
+                    let mut ret: i32 = -libc::EAGAIN;
+                    add_process(&mut ret as *mut i32, container_key, runc_process.pid);
+                    check_uprobe_ret(ret)?;
+                }
+            }
+            ContainerAction::Create => {
+                let container_id = container_id_o.unwrap();
```

Comment:
Are we ok with the risk of unwrapping here?

Reply:
Every `runc create` and `runc delete` command has to come with container ID. I never got any panic here. But on the other hand, maybe returning some descriptive error could be better here, just in case...

Reply:
Whatever you think is best! I usually like to wrap an error or return it since panic traces can be a bit harder to deal with